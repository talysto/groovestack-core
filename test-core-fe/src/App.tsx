import { Admin, Resource } from 'react-admin'
import { Box } from '@mui/material'

// Source Code import
import { Auth } from '@groovestack/auth'
import { Comments } from '@groovestack/comments'
import { 
        // HomeView,
        GroovestackDash } from '@groovestack/config'
import { Webhooks } from '@groovestack/webhooks'

import { mockDataProvider, mockAuthProvider, credentials, 
          //defaultAppConfig
        } from './data/mock-providers'
import { Company } from './resources/company'

// import { pkg as CoreBasePkg } from '@groovestack/base'
// import { pkg as CoreJobsPkg } from '@groovestack/jobs'
// import { pkg as CoreWebhooksPkg } from '@groovestack/webhooks'

const authProvider = await mockAuthProvider() // await Auth.Providers.Mock(params)

const appInit = true

const appConfig = { 
  has_admins: true, 
  user_roles: ['admin'], 
  oauth_providers: { 
    enabled: [
      {k: 'google', path: 'users/auth/google'},
      {k: 'apple', path: 'users/auth/apple'},
    ],
    configured: [
      {k: 'google', path: 'users/auth/google'},
      {k: 'apple', path: 'users/auth/apple'},
    ]
  },
  auth_providers: { 
    enabled: [
      {k: 'google', path: 'users/auth/google'},
      {k: 'apple', path: 'users/auth/apple'},
      {k: 'email' }
    ],
    configured: [
      {k: 'google', path: 'users/auth/google'},
      {k: 'apple', path: 'users/auth/apple'},
      {k: 'email' }
    ]
  }
}

const AppInitHeadline = () => {
  return (
    <Box sx={{ p: 3 }}>
      <div>There are currently no registered users on your application.</div>
      <div>Be the first!</div>
    </Box>
  )
}

const LoginPage = (props: any) => {
  return (
    <Auth.RA.LoginPage 
      {...props} 
      credentials={credentials} 
      appInit={appInit} 
      Headline={AppInitHeadline} 
    />
  )
}

function AdminApp() {
  // credentials.setAppConfig(defaultAppConfig)
  credentials.setAppConfig(appConfig)

  return (
    <Admin
      disableTelemetry
      dataProvider={mockDataProvider}
      authProvider={authProvider}
      loginPage={LoginPage}
      dashboard={GroovestackDash}
      layout={Auth.RA.Layout}
      requireAuth
      darkTheme={undefined}
      // theme={houseLightTheme}
    >
      <Resource
        name={Auth.Users.Name}
        icon={Auth.Users.Icon}
        // edit={User.Edit}
        list={Auth.Users.List}
        show={Auth.Users.Show}
        recordRepresentation="Auth Name"
      />

      <Resource
        name="Company"
        icon={Company.Icon}
        edit={Company.Edit}
        list={Company.List}
        create={Company.Create}
        recordRepresentation="name"
      />

      <Resource
        name="Comment"
        icon={Comments.Icon}
        list={Comments.List}
        edit={Comments.Edit}
      />

      <Resource name="JobLocker" />
      <Resource name="JobStat" />

      <Resource
        name="Webhook"
        icon={Webhooks.Icon}
        list={Webhooks.List}
        show={Webhooks.Show}
        recordRepresentation={Webhooks.recordRepresentation}
      />
    </Admin>
  )
}

export default AdminApp