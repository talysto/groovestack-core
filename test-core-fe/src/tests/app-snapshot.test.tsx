// import { TestShow } from '@/resources/company/CompanyList';
// import { AdminContext, testDataProvider, } from 'react-admin';
// import { TestShow } from '@/resources/company/CompanyList'
import App from '../App'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

test("Renders the Company List", () => {
  // render(<Users />)
  // render(
  //   <AdminContext>
  // <CompanyList />
  // </AdminContext>
  // )

  const { container } =
    render(
    // <AdminContext dataProvider={testDataProvider({
    //   //@ts-ignore
    //     getOne: () => Promise.resolve({ data: { id: 1, name: 'foo' } }),
    // })}>
    // <TestShow/>
        <App />
    ) 
    // </AdminContext>
      expect(container).toMatchSnapshot()

  
  expect(true).toBeTruthy()
})