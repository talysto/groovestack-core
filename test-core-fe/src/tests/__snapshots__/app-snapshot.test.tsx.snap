// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Company List 1`] = `
<div>
  <div
    class="css-g826q3-RaSimpleShowLayout-root"
  >
    <div
      class="MuiStack-root RaSimpleShowLayout-stack css-jfdv4h-MuiStack-root"
    >
      <span
        class="MuiStack-root ra-field ra-field-name RaSimpleShowLayout-row css-8l66u2-MuiStack-root-RaLabeled-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 RaLabeled-label css-1pnmrwp-MuiTypography-root"
        >
          <span>
            Name
          </span>
        </p>
        <span
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          foo
        </span>
      </span>
      <span
        class="MuiStack-root ra-field ra-field-address RaSimpleShowLayout-row css-8l66u2-MuiStack-root-RaLabeled-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 RaLabeled-label css-1pnmrwp-MuiTypography-root"
        >
          <span>
            Address
          </span>
        </p>
        <span
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          bar
        </span>
      </span>
    </div>
  </div>
</div>
`;
